### DISCUSSION TITLE
Is the signalling on the interchanges excessive?

### DISCUSSION THREAD

**Discussion by rpdelaney** [Opinionated-Blueprints/10-Books-Full-of-Rails#105] [General]
For example, consider the "Bottle" interchange:

![2024-12-04_13-17-43](https://github.com/user-attachments/assets/fda34353-5ff4-4d87-822b-23d9768442eb)

At least these signals can be deleted without breaking the sectioning:

![2024-12-04_13-17-43-2](https://github.com/user-attachments/assets/479a28fe-b401-47d9-8276-cf81da1df5b6)

Of course I wouldn't argue that the resource savings of a few rail signals is significant, because it's not. As far as I know, this isn't a big deal. Regardless, is there a specific reason for the density of the signals in these interchanges?

**Comment by Bocian-1** [comment]
These signals may not be strictly necessary for 6-car trains, but are very useful for 3- and 2-car trains by dividing single 6-car buffers into 2 3-car, or 3 2-car ones, and I think it's better to offer a single set of intersections with some signals that may not come into play rather than bloat the book with 3 separate sets, each optimised for a different train length.
I've also tested some intersections with and without these for 6-car trains and the improvements with the additional signals were significant as well. The 4-way I tested went from 84 -> 100 score

**Reply by brandon-gradient** [reply]
I think trains try to do a bit of look ahead in-game and "reserve" the next section (sections?) they will soon be on. So having a couple smaller sections leading up to an intersection (crossing, merge, split) could possibly help by not having trains too far away from the intersection claim the section. Even if that's all BS, it sounds like Bocian has some evidence that the is an effect so maybe it's not worth thinking too hard about.

**Reply by bwoods89** [reply]
I don't know if a look-ahead is happening in the game, but what I imagine is happening with the extra signal comes into play in high traffic scenarios. Imagine you have all the signals marked with red removed, and you have trains waiting to enter the intersection. Without the signals in red, a train has to wait all the way back at the entrance of the intersection until a conflicting train clears the now much bigger rail block ahead of them before the waiting train can advance at all. With the extra signals, the blocks or smaller, so the waiting train can start moving much sooner. This increases throughput as the train is stopped for less time.
